main.tf

```diff
@@ -197,6 +204,58 @@ locals {
   need_nat_ami_id     = local.nat_instance_enabled && length(var.nat_instance_ami_id) == 0
   nat_instance_ami_id = local.need_nat_ami_id ? data.aws_ami.nat_instance[0].id : try(var.nat_instance_ami_id[0], "")
 
+  # Locals for outputs
+  az_private_subnets_map = { for z in local.vpc_availability_zones : z => (
+    [for s in aws_subnet.private : s.id if s.availability_zone == z])
+  }
+
+  az_public_subnets_map = { for z in local.vpc_availability_zones : z => (
+    [for s in aws_subnet.public : s.id if s.availability_zone == z])
+  }
+
+  az_private_route_table_ids_map = { for k, v in local.az_private_subnets_map : k => (
+    [for t in aws_route_table_association.private : t.route_table_id if contains(v, t.subnet_id)])
+  }
+
+  az_public_route_table_ids_map = { for k, v in local.az_public_subnets_map : k => (
+    [for t in aws_route_table_association.public : t.route_table_id if contains(v, t.subnet_id)])
+  }
+
+  named_private_subnets_map = { for i, s in var.subnets_per_az_names : s => (
+    compact([for k, v in local.az_private_subnets_map : try(v[i], "")]))
+  }
+
+  named_public_subnets_map = { for i, s in var.subnets_per_az_names : s => (
+    compact([for k, v in local.az_public_subnets_map : try(v[i], "")]))
+  }
+
+  named_private_route_table_ids_map = { for i, s in var.subnets_per_az_names : s => (
+    compact([for k, v in local.az_private_route_table_ids_map : try(v[i], "")]))
+  }
+
+  named_public_route_table_ids_map = { for i, s in var.subnets_per_az_names : s => (
+    compact([for k, v in local.az_public_route_table_ids_map : try(v[i], "")]))
+  }
+
+  named_private_subnets_stats_map = { for i, s in var.subnets_per_az_names : s => (
+    [
+      for k, v in local.az_private_route_table_ids_map : {
+        az             = k
+        route_table_id = try(v[i], "")
+        subnet_id      = try(local.az_private_subnets_map[k][i], "")
+      }
+    ])
+  }
+
+  named_public_subnets_stats_map = { for i, s in var.subnets_per_az_names : s => (
```

Comment:
Does this PR miss public.tf and private.tf files change? I was going to validate whether you do pass the new subnet names as tags so they are available for future discovery, but didn't find it